state/constraints.go

```diff
@@ -32,6 +32,13 @@ type constraintsDoc struct {
 	Zones          *[]string
 }
 
+// constraintsDocRead is the mongodb representation of a constraints.Value.
+// including the DocID as we do not want to use this doc for updating the immutable _id.
```

Comment:
I added this, because adding this to the intial struct led to lot of regressions. Mostly in regards of overriding the existing `_id` with the new added `DocID`. 

I suppose `newConstraintsDoc(const constraints.Value)` returns a constrainsDoc, with DocID being "".

A possible solution, which I did not test yet, was to explicitly adding the DocID  to the doc during `newConstraintsDoc...`

Reply:
I am using another already defined type with inline and nested struct.